### PR TITLE
Tests: port lib/plugins to single test runner

### PR DESCRIPTION
**Pull request by jhnstn**
part of #3942 

Each test works in isolation

**Testing**

    npm run test-client
    npm run test-client client/lib/plugins/test/actions -- -w
    npm run test-client client/lib/plugins/test/log-store -- -w
    npm run test-client client/lib/plugins/test/store -- -w
    npm run test-client client/lib/plugins/test/utils -- -w
   npm run test-client client/lib/plugins/wporg-data/test/actions -- -w
  npm run test-client client/lib/plugins/wporg-data/test/list-store -- -w

cc @gziolo , @blowery

### CODE REVIEW

**Comment by jhnstn** [comment]
adding plugins/wporg-data to this PR

File: client/lib/plugins/wporg-data/test/list-store.js [hunk 1]
```diff
@@ -1,279 +1,258 @@
...
 			resetListsStore();
 			PluginsListsStore.getShortList( 'popular' );
-			assert.equal( localStorageMock.getActivity().set, 1 );
+			assert.isTrue( localStorageSpies.set.called );
 		} );
 
-		it( 'should not fetch from wporg if the category list is already in storage and is not yet after its TTL', function() {
-			Dispatcher.handleServerAction( actionsData.fetchedPopularPluginsList );
+		it( 'should not fetch from wporg if the category list is already in storage and is not yet after its TTL', () => {
 			resetListsStore();
```
Comment (jhnstn) [inline client/lib/plugins/wporg-data/test/list-store.js]: I'm not sure if the previous mock setup was incorrect but after converting to spies `fetchPluginsList` was being called if `resetListsStore` was called after the dispatch server action. That does seem to be the correct behavior where the `RESET_WPORG_PLUGINS_LIST` event clears out the storage.

**Comment by gziolo** [comment]
It need to be rebased :)

I noticed some issues with the following test file:
`time npm run test-client client/lib/plugins/test/store -- -w`

It takes over 13 seconds to execute:
> real    0m13.580s
user    0m3.075s
sys     0m0.523s

Any ideas?

File: client/lib/plugins/wporg-data/test/list-store.js (hunk 1 above)
Comment (gziolo) [inline client/lib/plugins/wporg-data/test/list-store.js]: It might change because you removed mockery call to replace `lodash/debounce` with identity.

**Comment by gziolo** [comment]
Other than that all other test files works nicely. I just skimmed the diff, because it's so huge. I'll get another look later today :)

File: client/lib/plugins/wporg-data/test/mocks/actions.js [hunk 2]
```diff
@@ -0,0 +1,8 @@
+/**
+ * Needed for store test
+ */
+import sinon from 'sinon';
```
Comment (gziolo) [inline client/lib/plugins/wporg-data/test/mocks/actions.js]: You can also use to save some keystrokes :)
```js
import { spy } from 'sinon';
```

File: client/lib/plugins/wporg-data/test/mocks/local-store.js [hunk 3]
```diff
@@ -0,0 +1,14 @@
+
```
Comment (gziolo) [inline client/lib/plugins/wporg-data/test/mocks/local-store.js]: Thanks for refactoring those mocks. They look way much cleaner now, and also tests are easier to read 👍

**Comment by gziolo** [comment]
Great work on that PR, lots of really nice refactorings 💯 

We need to address slow tests issue and we should be good to go.

You can also experiment with removed `debounce` mock. Personally I'd rather play with fake timers in the tests instead of changing production logic. On the other hand if tests are really dependant on that better leave it as is. We have more tests to move :)

**Comment by blowery** [comment]
Looks like `npm run test-client client/lib/plugins/test/store -- -w` just sits for a while after completing the tests. Guessing there's something async going set up that's waiting around to time out?

**Comment by blowery** [comment]
fixed, rebased on master, forced

**Comment by blowery** [comment]
Makes me think we need something to report on setTimeout and setInterval calls from within a test. I hacked this up in test/boot-test.js to find the problem in this PR:

```
const _setTimeout = global.setTimeout;
global.setTimeout = function logSetTimeout() { 
  console.error( 'setTimeout:', arguments );
  return _setTimeout.apply( global, arguments );
}
```

**Comment by blowery** [comment]
@gziolo could you give this another look? I'm recusing myself from reviewing it as I have a commit in it now

**Comment by gziolo** [comment]
Magic trick, issue solved 🚢 
We should investigate our options to mitigate issues around delayed async executions. I totally agree :)

**Pull request merged**